core/services/subscription_test.go

```diff
@@ -292,6 +292,9 @@ func TestServices_StartJobSubscription_RunlogNoTopicMatch(t *testing.T) {
 			require.NoError(t, store.CreateJob(&job))
 
 			runManager := new(mocks.RunManager)
+			// TODO: Is it supposed to do this?
+			runManager.On("CreateErrored", mock.Anything, mock.Anything, mock.Anything).
```

Comment:
Note that this change uncovered what I think was a bug - that we weren't calling `CreateErrored` in the event that we failed to parse logs.

Pretty sure this new behaviour is the intended one, but I'd need someone with more familiarity to confirm.

Reply:
Did you make a corresponding change to the code where logs are parsed to trigger CreateErrored? I think the implication for this are that we could fill the db up with errored runs, so I'm not sure it's desirable.

Reply:
The refactoring causes log parsing errors to be treated the same way as all the other errors.

Reply:
I believe that is quite deliberate, due to the above mentioned DB disk space concerns.